File: Bai_103/main.py
def isDistinct(num):
    Save = [0] * 256
    numstr = str(num)
    for c in numstr:
        if c == '0': return False
        if Save[ord(c)] > 0: return False
        Save[ord(c)] += 1
    return True

if __name__ == "__main__":
    with open('SOBATLAP.INP','r') as ifile: 
        with open("SOBATLAP.OUT", "w") as ofile:
            for line in ifile:
                N = int(line.strip())
                res = N
                while not isDistinct(res):
                    res += 1
                ofile.write(f"{res}\n")

File: Bai_103/main.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

bool isDistinct(int num){
    int Save[256] = {0};
    string numStr = to_string(num);
    for(char c : numStr){
        if(c == '0') return false;
        if(Save[c] > 0) return false;
        Save[c]++;
    }
    return true;
}

int main() {
    ifstream inputFile("SOBATLAP.INP");
    ofstream outputFile("SOBATLAP.OUT");
    
    int N;
    while(inputFile >> N){
        int result = N;
        while(!isDistinct(result))
            result++;
        outputFile << result << endl;
    }

    inputFile.close();
    outputFile.close();
    return 0;
}
